### ISSUE TITLE
[Question]why can this function always return a number

### ISSUE THREAD

**Issue by swr1bm86** [Fanael/rainbow-identifiers#4]
i have a little question about the implementation.

https://github.com/Fanael/rainbow-identifiers/blob/master/rainbow-identifiers.el#L85

why is the `rainbow-identifiers--hash-function` function always return a number?
cause hash generated use sha1 contains characters and integers, how does this algorithm avoid to pick up a character from the sha1 hash?

**Comment by Fanael** [comment]
The value returned from `(secure-hash 'sha1 … t)` is a 160-bit byte string, i.e. a string of 20 bytes from in the range [0, 255]. `rainbow-identifiers--hash-function` reads `rainbow-identifiers--hash-bytes-to-use` bottom-most bytes (in practice, 4 or 8, depending on Emacs integer range) as a big-endian integer and returns that.

Note that, despite 4 or 8 full bytes being used, Emacs tagging makes the effective integer size less than 32 or 64 bits, in practice being closer to 30 or 62 bits.

So, basically, `rainbow-identifiers--hash-function` takes 30 or 62 bits of the full 160-bit SHA-1 and interprets them as a big-endian integer.

Hope that helps.

**Comment by swr1bm86** [comment]
ok, i get it. aref just return the ASCII of a char.

https://www.gnu.org/software/emacs/manual/html_node/elisp/Array-Functions.html#Array-Functions

my bad, i should read doc more carefully.

thx @Fanael :)
